gigascience/gigadb-website
# Avoid double pagination in dataset url

# Pull request for issue: #1783

This is a pull request for the following functionalities:

* Avoid urls such as `dataset/view/id/100894/Samples_page/9/Files_page/4` for the case where two different paginations are triggered

## How to test?

* Navigate to http://gigadb.gigasciencejournal.com/dataset/view/id/100035
* Trigger samples pagination and confirm that the url has a Samples_page param
* Trigger files pagination and confirm that the url has no Samples_page and has Files_page param
* trigger again samples pagination and confirm that url has Samples_page param and not Files_page param

## How have functionalities been implemented?

The component SiteLinkPager createPageUrl method was overriden to avoid double pagination: both params are unset manually and then the correct param is set depending on what pagination was triggered

It should be okay to hard code the params unset because this component is only used for this one use case. Even if this component is resued for other use cases in the future, it will most likely work out of the box because the right param is always set after the manual unset

## Any issues with implementation?

--

## Any changes to automated tests?

--
